ecsmanage/management/commands/ecsmanage.py

```diff
@@ -0,0 +1,210 @@
+        """
+        if getattr(settings, 'ECSMANAGE_ENVIRONMENTS') is None:
+            raise CommandError(
+                'ECSMANAGE_ENVIRONMENTS was not found in the Django settings.'
+            )
+
+        ecs_configs = settings.ECSMANAGE_ENVIRONMENTS.get(self.env, None)
+        if ecs_configs is None:
+            raise CommandError(
+                f'Environment "{self.env}" is not a recognized environment in '
+                'ECSMANAGE_ENVIRONMENTS (environments include: '
+                f'{settings.ECSMANAGE_ENVIRONMENTS.keys()})'
+            )
+
+        config = {
+            'TASK_DEFINITION_NAME': '',
+            'CLUSTER_NAME': '',
+            'SECURITY_GROUP_TAGS': '',
+            'SUBNET_TAGS': '',
+            'LAUNCH_TYPE': 'FARGATE',
+            'AWS_REGION': 'us-east-1',
+        }
+
+        for config_name, config_default in config.items():
+            if ecs_configs.get(config_name) is None:
+                if config_default == '':
+                    raise CommandError(
+                        f'Environment "{self.env}" is missing required config '
+                        f'attribute {config_name}'
+                    )
+                else:
+                    config[config_name] = config_default
+            else:
+                config[config_name] = ecs_configs[config_name]
+
+        return config
+
+    def parse_response(self, response, key, idx=None):
+        """
+        Perform a key-value lookup on a response from the AWS API, wrapping it
+        in error handling such that if the lookup fails the response body
+        will get propagated to the end user.
+        """
+        if not response.get(key):
+            msg = f'Unexpected response from ECS API: {response}'
+            raise KeyError(msg)
+        else:
+            if idx is not None:
+                try:
+                    return response[key][0]
+                except (IndexError, TypeError):
+                    msg = (
+                        f"Unexpected value for '{key}' in response: "
+                        f'{response}'
+                    )
+                    raise IndexError(msg)
+            else:
+                return response[key]
+
+    def get_task_def(self, task_def_name):
+        """
+        Get the ARN of the latest ECS task definition for the app CLI.
+        """
+        task_def_response = self.ecs_client.list_task_definitions(
+            familyPrefix=task_def_name,
+            sort='DESC',
+            maxResults=1
+        )
+
+        return self.parse_response(task_def_response, 'taskDefinitionArns', 0)
+
+    def get_security_group(self, security_group_tags):
+        """
+        Get the ID of the security group to use for the app CLI.
+        """
+        filters = []
+        for tagname, tagvalue in security_group_tags.items():
+            filters.append({
+                'Name': f'tag:{tagname}',
+                'Values': [tagvalue]
+            })
```

Comment:
In Boto3, security groups and subnets have inconsistent interfaces for querying: security groups can be queried by name (although we don't typically assign them unique names) while subnets cannot. (Task definitions, in contrast, are generally uniquely identifiable by name.) See: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html

Given these constraints, I reasoned that the most flexible way to allow the user to specify the right security group/subnet was to require them to create a dictionary of tags to use as filters. This has the downside of requiring more configuration from the user and potentially not working for folks who don't use tags, but it has the upside of allowing the user to be as specific as possible about which resources they want to query.

Reply:
This sounds like a good compromise.